variables.tf

```diff
@@ -315,6 +315,22 @@ variable "ami_owners" {
   default     = ["amazon"]
 }
 
+variable "runner_ami_filter" {
+  description = "List of maps used to create the AMI filter for the Gitlab runner docker-machine AMI."
+  type        = "list"
+
+  default = [{
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-*"]
+  }]
+}
+
+variable "runner_ami_owners" {
+  description = "The list of owners used to select the AMI of Gitlab runner docker-machine instances."
+  type        = "list"
+  default     = ["Canonical"]
```

Comment:
A look up by name is not supported, I get the following error
```
Error: Error refreshing state: 1 error occurred:
	* module.runner.data.aws_ami.docker-machine: 1 error occurred:
	* module.runner.data.aws_ami.docker-machine: data.aws_ami.docker-machine: InvalidUserID.Malformed: Invalid user id: "Canonical"
	status code: 400, request id: 
```

Replacing Canonical by 099720109477 does the trick. Suggest that you replace Canonical by the id and add a comment line that it is the id of Canonical

Reply:
fixed in b08e1f8e5cb641c8d6ceb95cff2ce292bce1aee7